common/planners/src/plan_node_extras.rs

```diff
@@ -12,13 +12,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::collections::BTreeMap;
+
 use crate::Expression;
 
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum Projection {
+    /// column indices of the table
+    Columns(Vec<usize>),
+    /// inner column indices for tuple data type with inner columns.
+    /// the key is the column_index of ColumnEntry.
+    /// the value is the path indices of inner columns.
+    InnerColumns(BTreeMap<usize, Vec<usize>>),
```

Comment:
Seems the nested fields of the struct are already flattened, why do we keep track of this mapping?

Reply:
For inner columns, `column_index` does not exist in the table schema. We need to use `path_indices` to read them in the storage layer.

for example
```sql
create table test(id int, t tuple(a int, b int))
```

the table has the following columns

```
| name | index | path_indices |
| ---- | ----- | ------------ |
| id   | 0     | [0]          |
| t    | 1     | [1]          |
| t:a  | 2     | [1, 0]       |
| t:b  | 3     | [1, 1]       |
```
but the table schema only has fields `id` and `t`, if we want query `t:a` and `t:b`, we need to use path `path_indices`
`[1,0]` and `[1,1]` to read them in the storage layer